newsroom/navigations/service.py

```diff
@@ -9,12 +7,12 @@ class NavigationsService(NewshubAsyncResourceService[NavigationModel]):
 
     async def on_delete(self, doc: NavigationModel):
         """Remove references in products to this navigation entry once it is deleted"""
+        from newsroom.products import ProductsService
 
         await super().on_delete(doc)
 
         navigation = doc.id
-        products = get_resource_service("products").find(where={"navigations": str(navigation)})
-
-        for product in products:
+        products_cursor = await ProductsService().search({"navigations": navigation})
```

Comment:
@eos87 `ProductService` is being constructed once here and once for each product affected in the for loop. It would be good to construct it once, store in a local variable, and re-use that variable wherever you need to use the service